stdlib/public/core/Flatten.swift.gyb

```diff
@@ -313,40 +258,51 @@ public struct ${Collection}<
   /// reachable from `startIndex` by zero or more applications of
   /// `successor()`.
   public var endIndex: Index {
-    return ${Index}(_endIndexOfFlattened: _base)
+    return ${Index}(_base.endIndex, nil)
   }
 
   // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
   public func next(i: Index) -> Index {
-    fatalError("FIXME: swift-3-indexing-model implement")
+    // FIXME: swift-3-indexing-model: range checks?
```

Comment:
I think we don't need any range checks.  Base collections should be doing their own.